h2o-core/src/main/java/water/rapids/ast/prims/assign/AstRectangleAssign.java

```diff
@@ -265,22 +235,37 @@ private void assign_frame_scalar(Frame dst, int[] cols, AstNumList rows, String
     for (int i = 0; i < cols.length; i++)
       vecs2[i] = vecs[cols[i]];
     rows.sort();                // Side-effect internal sort; needed for fast row lookup
-    new AssignFrameStringScalarTask(rows, src).doAll(vecs2);
+    AssignFrameScalarTask.doAssign(rows, vecs2, src);
   }
 
-  private static class AssignFrameStringScalarTask extends RowSliceTask {
-    private String _src;
-    private AssignFrameStringScalarTask(AstNumList rows, String src) {
+  private static class AssignFrameScalarTask extends RowSliceTask {
```

Comment:
My suggestion here was the following: suppose you want to replace the first 25000 rows in a vec with value `42`. Also let's suppose those 25000 rows fit into 2.5 chunks. Then instead of iterating through 25000 rows, you could instead replace the first & second chunk with `ConstChunk(42)`, and then only iterate over the last 5000 rows.
The `Chunk` class has no way of offering this kind of optimization: by the time it can conclude you're writing a constant into the chunk, you've already wasted time on all the iterations...